rules/select_file.bzl

```diff
@@ -0,0 +1,36 @@
+"""
+select_file() build rule implementation.
+
+Selects a single file from the outputs of some target by given relative path.
+"""
+
+def _impl(ctx):
+    if ctx.attr.subpath and len(ctx.attr.subpath) == 0:
+        fail("Subpath can not be empty.")
+
+    out = None
+    canonical = ctx.attr.subpath.replace("\\", "/")
+    for file_ in ctx.attr.srcs.files.to_list():
+        if file_.path.replace("\\", "/").endswith(canonical):
+            out = file_
+            break
+    if not out:
+        files_str = ",\n".join([str(f.path) \
+for f in ctx.attr.srcs.files.to_list()])
```

Comment:
indentation?

Reply:
I was actually thinking that I can not use indentation if I used line continuation.
Apparently, I can :)
Corrected.